iroh-net/src/key.rs

```diff
@@ -17,14 +18,20 @@ pub use self::encryption::SharedSecret;
 use self::encryption::{public_ed_box, secret_ed_box};
 
 /// A public key.
-#[derive(Clone, Copy, Hash, PartialEq, Eq)]
+#[derive(Clone, Copy, PartialEq, Eq)]
 pub struct PublicKey {
     public: VerifyingKey,
     /// Cached version of `crypto_box::PublicKey` matching `public`.
     /// Stored as raw array, as `crypto_box::PublicKey` is not `Copy`.
     public_crypto_box: [u8; 32],
 }
 
+impl Hash for PublicKey {
```

Comment:
now that we determined this wasn't the issue, should we go back to using the derive `Hash`?

Reply:
no, this impl is still better, as it improves perf, and hashes the important piece